bluesky/spec_api.py

```diff
@@ -132,6 +135,9 @@ def ascan(motor, start, finish, intervals, time=None, *, md=None):
     md : dict, optional
         metadata
     """
+    if md is None:
+        md = {}
+    md = ChainMap(md, {'plan_name': 'ascan'})
```

Comment:
md has precedence over the `{}` added here so passed in values override default values so we would want to use `setdefault` instead of `update`.

It is also rude to mutate input.  Could have done this as

```python
local_md = {'plan_name': 'ascan'}
if md is not None:
    local_md.update(md)
md = local_md
```

but `ChainMap` is a bit more readable.